code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -75,92 +128,108 @@
+	owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out jets of flame!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, 1)
+	for(var/mob/living/carbon/C in ohearers(6, owner))
+		C.fire_stacks += 8
+		C.IgniteMob()
+	owner.fire_stacks = -20
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>[src] just makes [attack_text] worse by spewing fire on [owner]!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, 1)
+	owner.fire_stacks += 12
+	owner.IgniteMob()
+	return FALSE
 
 //Stealth
 
 /obj/item/clothing/suit/armor/reactive/stealth
 	name = "reactive stealth armor"
 	desc = "An experimental suit of armor that renders the wearer invisible on detection of imminent harm, and creates a decoy that runs away from the owner. You can't fight what you can't see."
+	cooldown_message = "<span class='danger'>The reactive stealth system activates, but is not charged enough to fully cloak!</span>"
+	emp_message = "<span class='warning'>The reactive stealth armor's threat assessment system crashes...</span>"
 
-/obj/item/clothing/suit/armor/reactive/stealth/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!does_react(hitby))
-		return FALSE
-	if(prob(hit_reaction_chance))
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive stealth system on [owner] activates, but is still recharging its holographic emitters!</span>")
-			return
-		var/mob/living/simple_animal/hostile/illusion/escape/E = new(owner.loc)
-		E.Copy_Parent(owner, 50)
-		E.GiveTarget(owner) //so it starts running right away
-		E.Goto(owner, E.move_to_delay, E.minimum_distance)
-		owner.alpha = 0
-		owner.visible_message("<span class='danger'>[owner] is hit by [attack_text] in the chest!</span>") //We pretend to be hit, since blocking it would stop the message otherwise
-		addtimer(VARSET_CALLBACK(owner, alpha, initial(owner.alpha)), 40)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
+	///when triggering while on cooldown will only flicker the alpha slightly. this is how much it removes.
+	var/cooldown_alpha_removal = 50
+	///cooldown alpha flicker- how long it takes to return to the original alpha
+	var/cooldown_animation_time = 3 SECONDS
+	///how long they will be fully stealthed
+	var/stealth_time = 4 SECONDS
+	///how long it will animate back the alpha to the original
+	var/animation_time = 2 SECONDS
+	var/in_stealth = FALSE
+
+/obj/item/clothing/suit/armor/reactive/stealth/cooldown_activation(mob/living/carbon/human/owner)
```

Comment:
this will actually never trigger since `reactivearmor_cooldown_duration` is 0 (default from parent)

the only reactive armor who actually has that var set is the teleport one (which also makes it the only one with an actual cooldown, the other ones can always be triggered because of this)

Reply:
default cooldown for all is now 5 seconds